backend/research/services/prospective.py

```
@@ -0,0 +1,714 @@
+            outcome_symbols,
+            start_date=min(basis_date for _, basis_date in evaluable),
+            end_date=datetime.now(TW_TIMEZONE).date(),
+        )
+        if evaluable and outcome_symbols
+        else {}
+    )
+
+    run_reports: list[dict[str, Any]] = []
+    resolved: list[dict[str, Any]] = []
+    valid_costs: set[tuple[float, float]] = set()
+    for record, evidence, basis_date, issues in prepared:
```

Ruff (0.16.0)

[warning] Loop control variable `evidence` not used within loop body

(B007)